broadinstitute/single_cell_portal_core
# Fixing broken dependency from react-scripts upgrade (SCP-4617)

#### BACKGROUND & CHANGES
Due to the upgrade for `react-scripts` in #1592, a transitive dependency for `autoprefixer` was omitted due to upgrades in other libraries.  While this didn't affect deployed environments, it shows up as spurious errors in CI, and prevented some local development in Docker due to `vite` not being able to load the missing dependency.  This update upgrades some related JS packages to restore the missing dependency.  

In addition, it reorders steps in the Dockerfile to allow faster iteration when updating startup scripts in Docker.  Additionally, it removes some broken/unneeded steps from `bin/run_tests.sh` that were obsoleted due to the Docker image upgrade from #1552.

#### MANUAL TESTING
1. Pull this branch
2. From the project root, rebuild the Docker image with:
```
docker build -t gcr.io/broad-singlecellportal-staging/single-cell-portal:development --no-cache .
```
3. Once the build is complete, start the container with the new image:
```
docker run --rm -it gcr.io/broad-singlecellportal-staging/single-cell-portal:development bash
```
4. Verify that the `autoprefixer` package was installed by checking the `node_modules` directory:
```
ls -l node_modules | grep autoprefixer
```
You should see something like `drwxr-xr-x   6 root root  4096 Aug 23 15:28 autoprefixer`
